linera-sdk/src/contract/runtime.rs

```diff
@@ -86,4 +110,64 @@ impl ContractRuntime {
             .authenticated_caller_id
             .get_or_insert_with(|| wit::authenticated_caller_id().map(ApplicationId::from))
     }
+
+    /// Retrieves the current system time, i.e. the timestamp of the block in which this is called.
+    pub fn system_time(&mut self) -> Timestamp {
+        *self
+            .timestamp
+            .get_or_insert_with(|| wit::read_system_timestamp().into())
+    }
+
+    /// Returns the current chain balance.
+    pub fn chain_balance(&mut self) -> Amount {
+        wit::read_chain_balance().into()
+    }
+
+    /// Returns the balance of one of the chain owners.
```

Comment:
I think also people who aren't chain owners can have a balance?

Reply:
Oops, my bad. I fixed the documentation.